test/parallel_api/numeric/numeric.ops/scan_2.pass.cpp

```diff
@@ -0,0 +1,350 @@
+    template <typename Policy, typename Iterator1, typename Size>
+    typename ::std::enable_if<
+        oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<Policy>::type>::value &&
+            is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value,
+        void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last,
+               Size n)
+    {
+        using KeyT = typename ::std::iterator_traits<Iterator1>::value_type;
+
+        TestingAlgoritm testingAlgo;
+
+        TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);
+
+        // Initialize source data in the buffer [keys_first, keys_last)
+        initialize_data(host_keys.get(), n);
+        const std::vector<KeyT> source_host_keys_state(host_keys.get(), host_keys.get() + n);
+
+        // Copy data from the buffer [keys_first, keys_last) to a device.
+        update_data(host_keys);
+
+        // Now we are ready to call tested algorithm
+        testingAlgo.call_onedpl(make_new_policy<new_kernel_name<Policy, 0>>(exec), keys_first, keys_last, keys_first);
+
+        retrieve_data(host_keys);
+
+        std::vector<KeyT> expected(n);
+        testingAlgo.call_serial(source_host_keys_state.cbegin(), source_host_keys_state.cend(), expected.data());
+        EXPECT_EQ_N(expected.cbegin(), host_keys.get(), n, testingAlgo.getMsg(true));
+    }
+#endif
+
+    // specialization for non-random_access iterators
+    template <typename Policy, typename Iterator1, typename Size>
+    typename ::std::enable_if<!is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value, void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last, Size n)
+    {
+    }
+};
+
+template <sycl::usm::alloc alloc_type, typename ValueType, typename BinaryOperation>
+void
+run_test()
+{
+    // Inclusive scan
+    {
+        // Non inplace
+        test2buffers<alloc_type, test_scan_non_inplace<ValueType, TestingAlgoritmInclusiveScan> >();
+        test2buffers<alloc_type, test_scan_non_inplace<ValueType, TestingAlgoritmInclusiveScanExt<BinaryOperation, 2 > > >();
+
+        // Inplace
+        test1buffer<alloc_type, test_scan_inplace<ValueType, TestingAlgoritmInclusiveScan>>();
+        test1buffer<alloc_type, test_scan_inplace<ValueType, TestingAlgoritmInclusiveScanExt<BinaryOperation, 2 > > >();
+    }
+
+    // Exclusive scan
+    {
+        // Non inplace
+        test2buffers<alloc_type, test_scan_non_inplace<ValueType, TestingAlgoritmExclusiveScan<2>>>();
+        test2buffers<alloc_type, test_scan_non_inplace<ValueType, TestingAlgoritmExclusiveScanExt<2, BinaryOperation > > >();
+
+        // Inplace
+        test1buffer<alloc_type, test_scan_inplace<ValueType, TestingAlgoritmExclusiveScan<2>>>();
+        test1buffer<alloc_type, test_scan_inplace<ValueType, TestingAlgoritmExclusiveScanExt<2, BinaryOperation>>>();
+    }
+}
+#endif // TEST_DPCPP_BACKEND_PRESENT
```

Comment:
It seems, that all code is guarded by TEST_DPCPP_BACKEND_PRESENT...
So, we dont test the host policies when DPCPP_BACKEND_PRESENT is 0?

Reply:
Fixed